src/main/java/com/esri/samples/na/closest_facility_static/ClosestFacilityStaticSample.java

```diff
@@ -0,0 +1,329 @@
+      ClosestFacilityTask closestFacilityTask = new ClosestFacilityTask("https://sampleserver6.arcgisonline.com/arcgis/rest/services/NetworkAnalysis/SanDiego/NAServer/ClosestFacility");
+
+      // create a table for facilities using the FeatureServer
+      FeatureTable facilitiesFeatureTable = new ServiceFeatureTable("https://services2.arcgis.com/ZQgQTuoyBrtmoGdP/ArcGIS/rest/services/San_Diego_Facilities/FeatureServer/0");
+      // create a feature layer from the table, apply facilities icon
+      FeatureLayer facilitiesFeatureLayer = new FeatureLayer(facilitiesFeatureTable);
+      facilitiesFeatureLayer.setRenderer(new SimpleRenderer(facilitySymbol));
+
+      // create a table for incidents using the FeatureServer
+      FeatureTable incidentsFeatureTable = new ServiceFeatureTable("https://services2.arcgis.com/ZQgQTuoyBrtmoGdP/ArcGIS/rest/services/San_Diego_Incidents/FeatureServer/0");
+      // create a feature layer from the table, apply incident icon
+      FeatureLayer incidentsFeatureLayer = new FeatureLayer(incidentsFeatureTable);
+      incidentsFeatureLayer.setRenderer(new SimpleRenderer(incidentSymbol));
+
+      // add the layers to the map
+      map.getOperationalLayers().addAll(Arrays.asList(facilitiesFeatureLayer, incidentsFeatureLayer));
+
+      // create the list to store the facilities
+      ArrayList<Facility> facilitiesList = new ArrayList<>();
+
+      // create the list to store the incidents
+      ArrayList<Incident> incidentsList = new ArrayList<>();
+
+      // wait for the feature layers to load to retrieve the facilities and incidents
+      facilitiesFeatureLayer.addDoneLoadingListener(() -> incidentsFeatureLayer.addDoneLoadingListener(() -> {
+        if (facilitiesFeatureLayer.getLoadStatus() == LoadStatus.LOADED && incidentsFeatureLayer.getLoadStatus() == LoadStatus.LOADED) {
+
+          // hide the progress indicator
+          progressIndicator.setVisible(false);
+
+          // zoom to the extent of the combined feature layers
+          Envelope fullFeatureLayerExtent = GeometryEngine.combineExtents(facilitiesFeatureLayer.getFullExtent(), incidentsFeatureLayer.getFullExtent());
+          mapView.setViewpointGeometryAsync(fullFeatureLayerExtent, 90);
+
+          // create query parameters to select all features
+          QueryParameters queryParameters = new QueryParameters();
+          queryParameters.setWhereClause("1=1");
+
+          // retrieve a list of all facilities
+          ListenableFuture<FeatureQueryResult> result = facilitiesFeatureTable.queryFeaturesAsync(queryParameters);
+          result.addDoneListener(() -> {
+            try {
+              FeatureQueryResult facilitiesResult = result.get();
+
+              // add the found facilities to the list
+              for (Feature facilityFeature : facilitiesResult) {
+                facilitiesList.add(new Facility(facilityFeature.getGeometry().getExtent().getCenter()));
```

Comment:
You could just cast the geometry to a Point since we know we are working with point features (could indicate this in a comment for the reader).

Reply:
If they were not point features, the closest point would probably be more useful than the center.

Reply:
Great shout - using `.getCenter()` did feel very clumsy!

Reply:
Sometimes we do use `getCenter` on the extent when we want to do something with the viewpoint, but that's really the only case where I've found it useful in samples thus far.